src/ResultWriter/WaveFieldWriter.cpp

```diff
@@ -280,9 +280,11 @@ void seissol::writer::WaveFieldWriter::init(unsigned int numVars,
 	// Set up for low order output flags
 	m_lowOutputFlags = new bool[WaveFieldWriterExecutor::NUM_LOWVARIABLES];
 	m_numIntegratedVariables = seissol::SeisSol::main.postProcessor().getNumberOfVariables();
+
 	for (size_t i = 0; i < WaveFieldWriterExecutor::NUM_PLASTICITY_VARIABLES; i++) {
-		m_lowOutputFlags[i] = (pstrain != 0L);
+		m_lowOutputFlags[i] = (pstrain != 0L) & (outputMask[numVars + i] != 0);
 	}
+
 	seissol::SeisSol::main.postProcessor().getIntegrationMask(&m_lowOutputFlags[WaveFieldWriterExecutor::NUM_PLASTICITY_VARIABLES]);
```

Comment:
Doesn't this just overwrite your mask?

Reply:
no, because it points the WaveFieldWriterExecutor::NUM_PLASTICITY_VARIABLES's element of the array, so will update only the `integral outputs`

Reply:
ok